src/components/CytoscapeGraph/CytoscapeGraph.tsx

```diff
@@ -333,15 +334,25 @@ export class CytoscapeGraph extends React.Component<CytoscapeGraphProps, Cytosca
       return;
     }
     const duration = this.props.graphDuration.value;
+    const healthPerNamespace = new Map<String, Promise<NamespaceHealth>>();
     // Asynchronously fetch health
     cy.nodes().forEach(ele => {
       const fqService = ele.data('service');
       if (fqService && (ele.data('isGroup') || !ele.data('parent'))) {
         const serviceParts = fqService.split('.');
+        if (serviceParts.length < 2) {
+          // Ignore health for special nodes such as "unknown"
+          return;
+        }
         const service = serviceParts[0];
         const namespace = serviceParts[1];
-        API.getServiceHealth(authentication(), namespace, service, duration).then(r => {
-          const health = r.data;
+        let promise = healthPerNamespace.get(namespace);
```

Comment:
I see in the code that promises are stored (for caching?), I guess to not repeat same query for other service in the same namespace.
If so, I would add a comment line, just to help others to fix it if something is wrong.

Reply:
Comment added (next to the map declaration)